internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -54,7 +54,7 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 
 		Importer: pluginsdk.ImporterValidatingResourceId(
 			func(id string) error {
-				_, err := commonids.ParseKubernetesClusterID(id)
+				_, err := commonids.ParseKubernetesClusterIDInsensitively(id)
```

Comment:
No need to parse it insensitively, because we require users to use a correct format to import it.